functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/jobs/JobActivityHistorySpec.groovy

```diff
@@ -0,0 +1,106 @@
+
+package org.rundeck.tests.functional.selenium.jobs
+
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.WebDriverWait
+import org.rundeck.util.annotations.SeleniumCoreTest
+import org.rundeck.util.container.SeleniumBase
+import org.rundeck.util.gui.pages.login.LoginPage
+import org.rundeck.util.gui.pages.execution.ExecutionShowPage
+import org.rundeck.util.gui.pages.jobs.JobListPage
+import org.rundeck.util.gui.pages.activity.ActivityPage
+import org.rundeck.util.common.jobs.JobUtils
+import java.time.Duration
+
+@SeleniumCoreTest
+class JobActivityHistorySpec extends SeleniumBase {
+    WebDriverWait wait
+    String jobId
+
+    def setupSpec() {
+        setupProject(SELENIUM_BASIC_PROJECT)
+    }
+
+    def setup() {
+        wait = new WebDriverWait(getDriver(), Duration.ofSeconds(30))
+        // Login
+        def loginPage = go LoginPage
+        loginPage.login(TEST_USER, TEST_PASS)
+        // Create a job
+        def jobDefinition = JobUtils.generateScheduledExecutionXml("TestJobActivityHistory")
+        def client = getClient() // Assuming a method exists to get RdClient
+        def response = JobUtils.createJob(SELENIUM_BASIC_PROJECT, jobDefinition, client)
+        assert response.succeeded.size() > 0: "Failed to create job"
+        jobId = response.succeeded[0].id
+        println "Job created with ID: ${jobId}"
+        // Run the job
+        def runResponse = JobUtils.executeJob(jobId, client)
+        assert runResponse.code() == 200: "Failed to execute job"
+        println "Job executed with response code: ${runResponse.code()}"
+    }
+
+    def cleanup() {
+        getDriver().quit()
+    }
+
+    /**
+     Test 1: Review Job Activity History from Job List Page
+     */
+    def "review job activity history from Job List Page"() {
+        when: "Navigate to the Job List Page"
+        def activityPage = go ActivityPage, SELENIUM_BASIC_PROJECT
+        activityPage.loadActivityPageForProject(SELENIUM_BASIC_PROJECT)
+        then: "Validate job execution is listed in Activity History"
+        activityPage.validatePage()
+        List<WebElement> activityList = activityPage.getActivityRows()
+        println "Activity List Size: ${activityList.size()}"
+        assert activityList.size() > 0: "No activity rows found"
+        WebElement firstActivityRow = activityList.get(0)
+        WebElement statusIcon = ExecutionShowPage.getActivityExecStatusIcon(firstActivityRow)
+        String status = statusIcon.getAttribute("data-statusstring")
+        println "First Activity Row Status: ${status}"
+        assert status.equalsIgnoreCase("SUCCEEDED"): "Expected 'SUCCEEDED', but found: '${status}'"
```

Comment:
What will ensure that the job has finished executing by the time this assertion is checked?

Reply:
I see now that the `wait.until` on line 56 will ensure this! 👍